controllers/shooter.py

```diff
@@ -27,65 +27,85 @@ class ShooterController(StateMachine):
 
     def __init__(self) -> None:
         # set default preference for goal height
-        pass
+        self.GoalHeight_Preference = GoalHeight.HIGH
 
     @state(first=True, must_finish=True)
     def preparing_intake(self) -> None:
         # send pitch and turret to angle
-
+        self.turret_component.set_angle(0.0)
         # if at pitch and turret angle
+        if self.turret_component.at_angle():
+            self.next_state("intaking")
         # next state intaking
-        pass
+        #pass
 
     @state(must_finish=True)
     def intaking(self) -> None:
-        # shooter intake
+        # shooter intake 
         # intake deploy
+        self.intake_component.deploy()
 
-        # if shooter is loaded
-        # retract intake
-        # next state tracking
-        pass
+        if self.shooter_component.is_loaded():
+            self.intake_component.retract()
+            self.next_state("tracking")
+        #pass
 
     @default_state
     def tracking(self) -> None:
         # if turret not indexed
         # return
+        if self.turret_component.find_index() is None:
```

Comment:
This is not correct, or necessary.
You need to work out what angle to point at the target, and set the `tilt` and `turret` angles to the calculated values, and the `shooter` flywheels to the correct speed.
Transition will happen if it is tracking those things properly, and the driver is pressing the shoot button.